lib/backend/dynamo/dynamodbbk.go

```diff
@@ -92,8 +92,18 @@ type Config struct {
 	// WriteTargetValue is the ratio of consumed write capacity to provisioned
 	// capacity. Required to be set if auto scaling is enabled.
 	WriteTargetValue float64 `json:"write_target_value,omitempty"`
+
+	// BillingMode sets on-demand capacity to the DynamoDB tables
+	BillingMode billingMode `json:"billing_mode,omitempty"`
 }
 
+type billingMode string
+
+const (
+	billingModeProvisioned billingMode = "provisioned"
+	billingModeOnDemand    billingMode = "on_demand"
+)
```

Comment:
nit: This feels close enough to the AWS convention of using `PROVISIONED` and `PAY_PER_REQUEST`, that it'll cause confusion.  IMO we should either use the exact same values for our representation, or to break from the aws API's pattern entirely and do something like `on_demand: true`.